run/cisco2john.pl

```diff
@@ -89,32 +89,42 @@ sub unique
     return !$uniq{$input}++;
 }
 
-# Credits for original code and description hobbit@avian.org,
-# SPHiXe, .mudge et al. and for John Bashinski
-# for Cisco IOS password encryption facts.
-#
-# Use of this code for any malicious or illegal purposes is strictly prohibited!
-#
-sub deobfuscate
-{
-    my @xlat = ( 0x64, 0x73, 0x66, 0x64, 0x3b, 0x6b, 0x66, 0x6f, 0x41,
-		 0x2c, 0x2e, 0x69, 0x79, 0x65, 0x77, 0x72, 0x6b, 0x6c,
-		 0x64, 0x4a, 0x4b, 0x44, 0x48, 0x53 , 0x55, 0x42 );
-    my ($ep) = @_;
-    my $dp = "";
-
-    if (!(length($ep) & 1)) {
-	    my ($s, $e) = ($ep =~ /^(..)(.+)/);
-	    $s = hex($s);
-	    for (my $i = 0; $i < length($e); $i += 2) {
-		    $dp .= sprintf "%c", hex(substr($e, $i, 2)) ^
-		      $xlat[$s++ % 26];
-	    }
-	    if ($dp =~ m/[\x00-\x19]/) {
-		    $dp = unpack("H*", $dp);
-	    }
-	    return $dp;
+sub cisco_decrypt {
+    ##################################################
+    # Nicked late 2021 from CPAN, Crypt::Cisco
+    # Copyright (C) Michael Vincent 2010, 2017
+    # www.VinsWorld.com
```

Comment:
Perhaps we also need to mention the CPAN code's license, not only copyright.

Reply:
OK... it says "This software is released under the same terms as Perl itself." 

https://dev.perl.org/licenses/

Reply:
So we should probably at least copy that statement.